### PR TITLE
[MRG] Updates for 7332732

### PR DESCRIPTION
**Pull request by HolterPhylo**

### CODE REVIEW

File: libMBIN/Source/NMS/GameComponents/GcMissionSequenceConstructSettlementBuildingWithScanEvent.cs [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
         /* 0x080 */ public NMSString0x80 MessageWhileBuilding;
         /* 0x100 */ public NMSString0x80 MessageWithItemsGathered;
-        /* 0x180 */ public NMSString0x20A ScanEvent;
-        /* 0x1A0 */ public float ForceCompleteSequenceAtStagePercentage;
-        /* 0x1A4 */ public NMSString0x80 DebugText;
+        /* 0x180 */ public NMSString0x80 MessageWhenDistant;
+        /* 0x200 */ public NMSString0x20A ScanEvent;
+        /* 0x220 */ public float ForceCompleteSequenceAtStagePercentage;
+        /* 0x224 */ public NMSString0x80 DebugText;
+        [NMS(Size = 0x4, Ignore = true)]
```
Comment (monkeyman192) [inline libMBIN/Source/NMS/GameComponents/GcMissionSequenceConstructSettlementBuildingWithScanEvent.cs]: This end padding is no longer required
Edit: Same with the other ones added

Comment (HolterPhylo) [reply libMBIN/Source/NMS/GameComponents/GcMissionSequenceConstructSettlementBuildingWithScanEvent.cs]: Curious, why TkModelResource and others but not these?  Or they don't need anymore?

Comment (chrispurnell) [reply libMBIN/Source/NMS/GameComponents/GcMissionSequenceConstructSettlementBuildingWithScanEvent.cs]: TkModelResource and friends have a missing field at the end of the structure.  They are special cases.
I also noticed you added unneeded end padding to GcAtlasSendSubmitBase and GcNPCDebugSpawnData in a previous commit.

Comment (HolterPhylo) [reply libMBIN/Source/NMS/GameComponents/GcMissionSequenceConstructSettlementBuildingWithScanEvent.cs]: Ok, I understand that TkModelResource is Size 0x84, field is 0x80 so we need to add 0x4 at the end.
Now, for example, GcMissionSequenceConstructSettlementBuildingWithScanEvent is Size 0x2A8, last field is 0x80 and ends at 0x2A4:  0x4 bytes are missing at the end, same as TkModelResource.  Please explain why it is different.  I surely miss something!

Comment (chrispurnell) [reply libMBIN/Source/NMS/GameComponents/GcMissionSequenceConstructSettlementBuildingWithScanEvent.cs]: GcMissionSequenceConstructSettlementBuildingWithScanEvent has an alignment of 8 but 0x2A4 is not a multiple of 8.
The need for end padding in this case was removed when I added code to realign the stream after the last field was read from or written to an mbin.

Comment (monkeyman192) [reply libMBIN/Source/NMS/GameComponents/GcMissionSequenceConstructSettlementBuildingWithScanEvent.cs]: We only need the padding bytes when there is space that is unaccounted for by both data and alignment.
Basically, there are almost no classes that need alignment any more other than some specific ones which already have it. So rule of thumb: Don't add alignment anywhere, and also don't remove it in places that there is alignment added.

Comment (HolterPhylo) [reply libMBIN/Source/NMS/GameComponents/GcMissionSequenceConstructSettlementBuildingWithScanEvent.cs]: " has an alignment of 8 but 0x2A4 is not a multiple of 8."  That is what I did not get.  Thanks

File: libMBIN/Source/NMS/GameComponents/GcMessageNPCBehaviourEvent.cs [hunk 2]
```diff
@@ -13,7 +13,5 @@ public class GcMessageNPCBehaviourEvent : NMSTemplate
         /* 0x20 */ public GcNPCTriggerTypes InteractionTrigger;
         /* 0x24 */ public int InteractionSubType;
         /* 0x30 */ public Vector3f Position;
-        [NMS(Size = 0x10, Ignore = true)]
-        /* 0x40 */ public byte[] EndPadding;  // this is really weird but there doesn't seem to be anything here and the size *is* 0x50...
```
Comment (monkeyman192) [inline libMBIN/Source/NMS/GameComponents/GcMessageNPCBehaviourEvent.cs]: These I think need it. As you can see from my comment. The class should be 0x50 bytes, however the data only takes up 0x40 of them... A lot of these message ones seem to have them and I have no idea why but we should keep them. (Same with the GcAtlasSendSubmitSettlement.cs potentially?

Comment (chrispurnell) [reply libMBIN/Source/NMS/GameComponents/GcMessageNPCBehaviourEvent.cs]: Yes, this, GcAtlasSendSubmitSettlement and GcMessagePetBehaviourEvent are all required.
It's almost certanly just a field that is not serialized in the mxml.

**Pull request merged**